File: 56-merge-intervals/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        if(intervals.size()<=1){
            return intervals;
        }
        vector<vector<int>> merged;
        sort(intervals.begin(),intervals.end());

        for(int i=1;i<intervals.size();i++){
            if(intervals[i][0] <= intervals[i-1][1]){
                intervals[i][0] = min(intervals[i][0], intervals[i-1][0]);
                intervals[i][1] = max(intervals[i][1], intervals[i-1][1]);
            }
            else{
                merged.push_back(intervals[i-1]);
            }
        }
        merged.push_back(intervals[intervals.size()-1]);
        return merged;
    }
};

File: 56-merge-intervals/merge-intervals.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """
        
        size = len(intervals)
        if (size < 2):
            return intervals
        
        res = []
        intervals.sort()
        i = 1

        while (i < size):
            

            if (intervals[i-1][1] >= intervals[i][0]):
                intervals[i][0] = min(intervals[i][0], intervals[i-1][0])
                intervals[i][1] = max(intervals[i][1], intervals[i-1][1])
            
            else:
                res.append(intervals[i-1])
            
            i += 1

        res.append(intervals[i-1])
        
        return res